integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/partition/TestDDLForPartitionTable.scala

```diff
@@ -122,6 +122,254 @@ class TestDDLForPartitionTable  extends QueryTest with BeforeAndAfterAll {
     intercept[Exception] { sql("alter table test drop columns(c)") }
   }
 
+  test("test exception if hash number is invalid") {
+    sql("DROP TABLE IF EXISTS test_hash_1")
+    try {
+      sql(
```

Comment:
the test cases would pass if the query does not throw an exception.
use intercept instead of try the above condition would be met automatically.
val exception: Exception = intercept[Exception] {sql(..)}
assert(exception.getMessage.equals(..))